src/unstract/sdk/utils/callback_manager.py

```diff
@@ -0,0 +1,123 @@
+import logging
+from typing import Callable, Optional, Union
+
+import tiktoken
+from llama_index.core.callbacks import (
+    CallbackManager as LlamaIndexCallbackManager,
+)
+from llama_index.core.callbacks import TokenCountingHandler
+from llama_index.core.embeddings import BaseEmbedding
+from llama_index.core.llms import LLM
+from transformers import AutoTokenizer
+
+from unstract.sdk.utils.usage_handler import UsageHandler
+
+logger = logging.getLogger(__name__)
+
+
+class CallbackManager:
+    """Class representing the CallbackManager to manage callbacks.
+
+    Use this over the default service context of llama index
+
+    This class supports a tokenizer, token counter,
+    usage handler, and  callback manager.
+
+    Attributes:
+        None
+
+    Methods:
+        set_callback_manager: Returns a standard callback manager
+
+    Example:
+        callback_manager = CallbackManager.
+                            set_callback_manager(
+                                llm="default",
+                                embedding="default")
+    """
+
+    @staticmethod
+    def set_callback_manager(
+        platform_api_key: str,
+        llm: Optional[LLM] = None,
+        embedding: Optional[BaseEmbedding] = None,
+        workflow_id: str = "",
+        execution_id: str = "",
+    ) -> LlamaIndexCallbackManager:
+        """Sets the standard callback manager for the llm. This is to be called
+        explicitly whenever there is a need for the callback handling defined
+        here as handlers is to be invoked.
+
+        Parameters:
+            llm (LLM): The LLM type
+
+        Returns:
+            CallbackManager tyoe of llama index
+
+        Example:
+            UNCallbackManager.set_callback_manager(
+                platform_api_key: "abc",
+                llm=llm,
+                embedding=embedding
+            )
+        """
+
+        if llm:
+            tokenizer = CallbackManager.get_tokenizer(llm)
+        elif embedding:
+            tokenizer = CallbackManager.get_tokenizer(embedding)
+
+        token_counter = TokenCountingHandler(tokenizer=tokenizer, verbose=True)
+        usage_handler = UsageHandler(
+            token_counter=token_counter,
+            platform_api_key=platform_api_key,
+            llm_model=llm,
+            embed_model=embedding,
+            workflow_id=workflow_id,
```

Comment:
I know this comes from some existing code, however what happens in case of usage from prompt studio?
CC: @johnyrahul @gaya3-zipstack